src/conn.rs

```diff
@@ -473,6 +504,32 @@ impl<'poll> Conn<'poll> {
     }
 
     fn handle_msg(&mut self, msg: Msg, evs: &mut Vec<ConnEv>, logger: &mut Logger) {
+        if let Msg {
+            cmd: Cmd::CAP { client: _, ref subcommand, ref params },
+            ..
+        } = msg
+        {
+            match subcommand.as_ref() {
+                "ACK" => {
+                    if params.iter().any(|cap| cap.as_str() == "sasl") {
+                        self.plain_sasl_authenticate();
+                    }
+                }
+                "NAK" => {
+                    self.end_capability_negotiation();
+                }
+                &_ => {}
+            };
+        }
+
+        match msg.cmd {
+            // SASL authentication successful or failed
```

Comment:
We should refer to https://ircv3.net/specs/extensions/sasl-3.1.html#numerics-used-by-this-extension here. 903: RPL_SASLSUCCESS, 904: ERR_SASLFAIL.

Reply:
changed